src/main/java/eu/europeana/downloads/ListIdentifiersQuery.java

```diff
@@ -97,37 +94,46 @@ public void execute(OAIPMHServiceClient oaipmhServer) {
 
     private void executeSingleThreadListRecord(OAIPMHServiceClient oaipmhServer, String setIdentifier) {
         List<String> identifiers = getIdentifiers(oaipmhServer, setIdentifier);
```

Comment:
For now it's fine, but to reduce complexity perhaps it's better to always do harvesting the same way (multi-threaded). If we only need to retrieve 1 set we can simply use the same approach but initialize the threadpool with 1 thread.

Reply:
made the changes. Initiated thread pool with one thread and downloading a single set with the same approach